src/super_gradients/common/sg_loggers/utils.py

```diff
@@ -0,0 +1,171 @@
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+
+import numpy as np
+
+
+@dataclass
+class PlottableMetricOutput:
```

Comment:
I think we should not make this class responsible for plotting itself.

We cannot see all the use-cases right now. Ex: may want to draw multiple lines on same plot (PR curve per each class). Or histograms or something else. For each of these use scenarios we would need to modify this class. 

Instead we should render the plot inside the metric as a regular image and return it `PlottableMetricOutput(value=image)`. So logic of generation of the image becomes responsibility of the Metric and not the class itself or logger.

Reply:
@BloodAxe after our discussion I changed the code accordingly. See new description and new code